### PR TITLE
JCLOUDS-660: Implement AzureBlob set ACL

### PR DESCRIPTION
**Pull request by andrewgaul**
Reference:

http://msdn.microsoft.com/en-us/library/azure/dd179391.aspx

### CODE REVIEW

File: providers/azureblob/src/test/java/org/jclouds/azureblob/AzureBlobClientTest.java [hunk 1]
```diff
@@ -232,6 +233,22 @@ public void testGetPublicAccessForContainer() throws SecurityException, NoSuchMe
...
+               "PUT https://identity.blob.core.windows.net/container?restype=container&comp=acl HTTP/1.1");
+      assertNonPayloadHeadersEqual(request,
+               "x-ms-blob-public-access: container\n" +
+               "x-ms-version: 2012-02-12\n");
+      assertPayloadEquals(request, null, null, false);
+
+      assertResponseParserClassEquals(method, request, ParseETagHeader.class);
+      assertSaxResponseParserClassEquals(method, null);
+      assertFallbackClassEquals(method, null);
+   }
```
Comment (nacx) [inline providers/azureblob/src/test/java/org/jclouds/azureblob/AzureBlobClientTest.java]: Add a test to verify the PRIVATE case does not add the header?

**Comment by nacx** [comment]
Thanks! Just one minor comment. Apart from that lgtm!

File: providers/azureblob/src/test/java/org/jclouds/azureblob/AzureBlobClientTest.java (hunk 1 above)
Comment (andrewgaul) [inline providers/azureblob/src/test/java/org/jclouds/azureblob/AzureBlobClientTest.java]: Done.

**Comment by andrewgaul** [comment]
Pushed to master as 35d6313e63a3f6b6b40dda262b54369dbaab437f.

**Pull request closed without merging**